src/protagonist/API/Features/CustomHeaders/Requests/DeleteCustomHeader.cs

```diff
@@ -0,0 +1,50 @@
+﻿using API.Infrastructure.Requests;
+using DLCS.Core;
+using DLCS.Model.Messaging;
+using DLCS.Repository;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace API.Features.CustomHeaders.Requests;
+
+public class DeleteCustomHeader: IRequest<ResultMessage<DeleteResult>>
+{
+    public DeleteCustomHeader(int customerId, string customHeaderId)
+    {
+        CustomerId = customerId;
+        CustomHeaderId = customHeaderId;
+    }
+    
+    public int CustomerId { get; }
+    
+    public string CustomHeaderId { get; }
+}
+
+public class DeleteCustomHeaderHandler : IRequestHandler<DeleteCustomHeader, ResultMessage<DeleteResult>>
+{
+    private readonly DlcsContext dbContext;
+
+    public DeleteCustomHeaderHandler(DlcsContext dbContext)
+    {
+        this.dbContext = dbContext;
+    }
+
+    public async Task<ResultMessage<DeleteResult>> Handle(DeleteCustomHeader request, CancellationToken cancellationToken)
+    {
+        var deleteResult = DeleteResult.NotFound;
```

Comment:
`deleteResult` only ever has 2 different values, is it cleaner to use use the enum value directly rather than having a mutating variable? Using a specific enum value at the place it's used saves looking up in the code to see the current value.

E.g., in these it's evident what the return value is:
```cs
if (customHeader == null) return new ResultMessage<DeleteResult>("CustomHeader not found", DeleteResult.NotFound);

return new ResultMessage<DeleteResult>("CustomHeader deleted", DeleteResult.Deleted);
```

whereas here we need to look around to see what the current value of `deleteResult` is
```cs
if (customHeader == null) return new ResultMessage<DeleteResult>(message, deleteResult);

return new ResultMessage<DeleteResult>(message, deleteResult);
```

Reply:
That's a good point, I've made it a little hard to follow. This has been changed in 95770330c2a1a080c057d35aab57ae96ae800260